pages/treeherder.py

```diff
@@ -28,16 +28,18 @@ class TreeherderPage(Base):
     _next_ten_locator = (By.CSS_SELECTOR, 'div.btn:nth-child(1)')
     _next_twenty_locator = (By.CSS_SELECTOR, 'div.btn:nth-child(2)')
     _next_fifty_locator = (By.CSS_SELECTOR, 'div.btn:nth-child(3)')
+    _resultset_visible_locator = (By.CSS_SELECTOR, 'span.hidden.ready')
     _quick_filter_locator = (By.ID, 'quick-filter')
     _repos_menu_locator = (By.ID, 'repoLabel')
     _result_sets_locator = (By.CSS_SELECTOR, '.result-set:not(.row)')
+    _resultset_visible_locator = (By.CSS_SELECTOR, 'span.hidden.ready')
     _unchecked_repos_links_locator = (By.CSS_SELECTOR, '#repoLabel + .dropdown-menu .dropdown-checkbox:not([checked]) + .dropdown-link')
     _unclassified_failure_count_locator = (By.ID, 'unclassified-failure-count')
     _unclassified_failure_filter_locator = (By.CSS_SELECTOR, '.btn-unclassified-failures')
 
     def wait_for_page_to_load(self):
-        self.wait.until(lambda s: self.unclassified_failure_count > 0)
-        return self
+        el = self.find_element(*self._resultset_visible_locator)
+        self.wait.until(EC.visibility_of(el))
```

Comment:
I've used this implementation and I've got `TimeoutException`, while I was trying to use `page.wait_for_page_to_load()`.

Did you have the same?